src/Microsoft.AspNetCore.SystemWebAdapters/HttpContext.cs

```diff
@@ -58,6 +58,32 @@ public IPrincipal User
 
     public DateTime Timestamp { get; } = DateTime.UtcNow.ToLocalTime();
 
+    public void RewritePath(string path)
+    {
+        if (path is null)
+        {
+            throw new ArgumentNullException(nameof(path));
+        }
+
+        // Extract query string
+        string? qs = null;
+        var iqs = path.IndexOf('?', StringComparison.Ordinal);
+
+        if (iqs >= 0)
+        {
+            qs = (iqs < path.Length - 1) ? path[iqs..] : string.Empty;
+            path = path[..iqs];
+        }
+
+        if (!path.StartsWith("/", StringComparison.Ordinal))
+        {
+            path = "/" + path;
+        }
+
+        _context.Request.QueryString = new QueryString(qs);
+        _context.Request.Path = path;
```

Comment:
Do we know if the input is expected to be escaped or un-escaped? Un-escaped is my guess, though it's weird that it could be un-escaped and include a query.
```suggestion
        _context.Request.Path = new PathString(path);
```

Reply:
Yeah, it seems weird, but it appears to be unescaped with a query. I constructed the unit tests from trying out various permutations against HttpContext